token-standard/cli/src/apis/ledger-api-utils.ts

```diff
@@ -322,3 +341,98 @@ function signTransaction(
     signedHash,
   };
 }
+
+interface Completion {
+  updateId: string;
+  // the openAPI definition claims these two can be null
+  synchronizerId?: string;
+  recordTime?: string;
+}
+
+const COMPLETIONS_LIMIT = 100;
+const COMPLETIONS_STREAM_IDLE_TIMEOUT_MS = 1000;
+/**
+ * Polls the completions endpoint until
+ * the completion with the given (userId, commandId, submissionId) is returned.
+ * Then returns the updateId, synchronizerId and recordTime of that completion.
+ */
+async function awaitCompletion(
+  ledgerClient: LedgerJsonApi,
+  ledgerEnd: number,
+  partyId: string,
+  userId: string,
+  commandId: string,
+  submissionId: string,
+): Promise<Completion> {
+  const responses = await ledgerClient.postV2CommandsCompletions(
+    {
+      userId,
+      parties: [partyId],
+      beginExclusive: ledgerEnd,
+    },
+    COMPLETIONS_LIMIT,
+    COMPLETIONS_STREAM_IDLE_TIMEOUT_MS,
+  );
+  const completions = responses.filter(
+    (response) => !!response.completionResponse.Completion,
+  );
+
+  const wantedCompletion = completions.find((response) => {
+    const completion = response.completionResponse.Completion;
+    return (
+      completion.value.userId === userId &&
+      completion.value.commandId === commandId &&
+      completion.value.submissionId === submissionId
+    );
+  });
+
+  if (wantedCompletion) {
+    const status = wantedCompletion.completionResponse.Completion.value.status;
+    if (status && status.code) {
```

Comment:
```suggestion
    if (status && status.code !== 0) {
```
maybe a bit clearer here?